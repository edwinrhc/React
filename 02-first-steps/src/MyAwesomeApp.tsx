import type {CSSProperties} from "react";

const firstName = 'Edwin';
const lastName = 'HC';

const favoriteGames = ['Dota 2', 'Star Craft 2', 'Half Life'];
const isActive = false;

const address = {
    zipCode: 12345,
    counter: 'Peru'
}

const myStyle: CSSProperties = {
    backgroundColor: '#fafafa',
    borderRadius: 10,
    padding: 10,
    marginTop: 20
}

export function MyAwesomeApp() {

    return (
        <>
            <div>
                <h1 data-testid="first-name-title">{firstName}</h1>
                <h2>{lastName}</h2>
                <p className="mi-clave-favorita">{favoriteGames.join(', ')}</p>
                <h1>{isActive ? 'Activo' : 'No activo'}</h1>
                <p style={myStyle}>
                    {JSON.stringify(address)}
                </p>
            </div>
        </>
    )
}
