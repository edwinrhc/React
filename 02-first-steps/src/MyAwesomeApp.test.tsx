import {describe, expect, test} from "vitest";
import {render, screen} from "@testing-library/react";
import {MyAwesomeApp} from "./MyAwesomeApp.tsx";

describe('MyAwesomeApp', () => {
    test('should render  firstName and lastName', () => {
        const {container} = render(<MyAwesomeApp/>);
        const h1 = container.querySelector('h1');
        const h2 = container.querySelector('h2');

        expect(h1?.innerHTML).toContain('Edwin');
        expect(h2?.innerHTML).toContain('HC');
    });

    test('should render  firstName and lastName - Screen', () => {
        render(<MyAwesomeApp />);
        screen.debug();
        console.log('---------------------------------------------------')
        const h1 = screen.getByTestId('first-name-title');
        expect(h1.innerHTML).toContain('Edwin');
    });

    // Fotografia de la pantalla  como test
    test('should match snapsho', () => {
        const { container } = render(<MyAwesomeApp />);
        expect(container).toMatchSnapshot();
    })
});
