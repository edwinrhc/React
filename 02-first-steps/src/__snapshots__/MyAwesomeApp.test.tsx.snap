// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapsho 1`] = `
<div>
  <div>
    <h1
      data-testid="first-name-title"
    >
      Edwin
    </h1>
    <h2>
      HC
    </h2>
    <p
      class="mi-clave-favorita"
    >
      Dota 2, Star Craft 2, Half Life
    </p>
    <h1>
      No activo
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 10px; padding: 10px; margin-top: 20px;"
    >
      {"zipCode":12345,"counter":"Peru"}
    </p>
  </div>
</div>
`;
